crates/sui/src/client_commands.rs

```diff
@@ -3098,3 +3107,66 @@ async fn check_protocol_version_and_warn(client: &SuiClient) -> Result<(), anyho
 
     Ok(())
 }
+
+/// Fetch move packages based on the provided package IDs.
+async fn fetch_move_packages(
+    client: &SuiClient,
+    package_ids: &[ObjectID],
+) -> Result<Vec<MovePackage>, anyhow::Error> {
+    let objects = client
+        .read_api()
+        .multi_get_object_with_options(package_ids.to_vec(), SuiObjectDataOptions::bcs_lossless())
+        .await
+        .map_err(|e| anyhow!("{e}"))?;
+
+    objects
+        .into_iter()
+        .map(|o| {
+            let o = o.into_object().map_err(|e| anyhow!("{e}"))?;
```

Comment:
I think you can avoid the format string entirely -- part of me is curious to see what happens if you omitted the `map_err` entirely and just used `?` but I'm guessing you tried that already.

```suggestion
            let o = o.into_object().map_err(|e| anyhow!(e))?;
```